src/datatypes2/src/types/timestamp_type.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright 2022 Greptime Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use arrow::datatypes::{
+    DataType as ArrowDataType, TimeUnit as ArrowTimeUnit,
+    TimestampMicrosecondType as ArrowTimestampMicrosecondType,
+    TimestampMillisecondType as ArrowTimestampMillisecondType,
+    TimestampNanosecondType as ArrowTimestampNanosecondType,
+    TimestampSecondType as ArrowTimestampSecondType,
+};
+use common_time::timestamp::TimeUnit;
+use common_time::Timestamp;
+use enum_dispatch::enum_dispatch;
+use paste::paste;
+use serde::{Deserialize, Serialize};
+use snafu::OptionExt;
+
+use crate::data_type::ConcreteDataType;
+use crate::error;
+use crate::prelude::{
+    DataType, LogicalTypeId, MutableVector, ScalarVectorBuilder, Value, ValueRef, Vector,
+};
+use crate::timestamp::{
+    TimestampMicrosecond, TimestampMillisecond, TimestampNanosecond, TimestampSecond,
+};
+use crate::types::LogicalPrimitiveType;
+use crate::vectors::{
+    PrimitiveVector, TimestampMicrosecondVector, TimestampMicrosecondVectorBuilder,
+    TimestampMillisecondVector, TimestampMillisecondVectorBuilder, TimestampNanosecondVector,
+    TimestampNanosecondVectorBuilder, TimestampSecondVector, TimestampSecondVectorBuilder,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[enum_dispatch(DataType)]
+pub enum TimestampType {
+    TimestampSecondType(TimestampSecondType),
+    TimestampMillisecondType(TimestampMillisecondType),
+    TimestampMicrosecondType(TimestampMicrosecondType),
+    TimestampNanosecondType(TimestampNanosecondType),
```

Comment:
What about using more concise variant names?
```suggestion
    Second(TimestampSecondType),
    Millisecond(TimestampMillisecondType),
    Microsecond(TimestampMicrosecondType),
    Nanosecond(TimestampNanosecondType),
```